pkg/pool-stable-phantom/contracts/StablePhantomPool.sol

```diff
@@ -442,151 +442,140 @@ contract StablePhantomPool is IRateProvider, BaseGeneralPool, ProtocolFeeCache {
         uint256 indexIn,
         uint256 indexOut,
         uint256[] memory scalingFactors
-    ) private returns (uint256) {
+    ) private returns (uint256 downscaledAmountCalculated) {
         _upscaleArray(balances, scalingFactors);
 
         (uint256 virtualSupply, uint256[] memory balancesWithoutBpt) = _payProtocolFeesBeforeJoinExit(balances);
 
-        bool bptIsTokenIn = swapRequest.tokenIn == IERC20(this);
+        bool isJoinSwap = swapRequest.tokenOut == IERC20(this);
+        bool isGivenIn = swapRequest.kind == IVault.SwapKind.GIVEN_IN;
+
+        uint256 upscaledAmountGiven = _upscale(swapRequest.amount, scalingFactors[isGivenIn ? indexIn : indexOut]);
```

Comment:
Nitpicky, but I'd move this upscale up a few lines to have it next to the balances upscaling.

Reply:
(conceptually I think of this as two functions - one that deals with upscaling/downscaling, extracting data from `request`, etc., and one that handles the actual join/exit)